plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -317,7 +316,7 @@ public PostgreSqlClient(
                 .addDefaultRules()
                 // TODO allow all comparison operators for numeric types
                 .add(new RewriteComparison(RewriteComparison.ComparisonOperator.EQUAL, RewriteComparison.ComparisonOperator.NOT_EQUAL))
-                .add(new RewriteLike())
+                .map("$like_pattern(value: varchar, pattern: varchar): boolean").to("value LIKE pattern")
```

Comment:
Since the API is entirely internal and meant to be used by connectors, can we get away with writing the AST manually in java instead of defining an antlr grammar/using a visitor to create the AST? 

In my opinion, this doesn't seem too bad to manually write. But maybe others feel differently 🤷‍♀️ 
```
new CallPattern(
                        "$like_pattern",
                        List.of(
                                new ExpressionCapture("value", type("varchar")),
                                new ExpressionCapture("pattern", type("varchar"))),
                        Optional.of(type("boolean"))
```

Reply:
Maybe adding a test case for a more complex example would show the benefits of the DSL?

Reply:
The DSL syntax here is similar to SQL, so hopefully feels familiar, with notable differences:

- call parameters have explicit types, call result has optional explicit type (this is central for function pattern matching)
- $ sign is allowed in identifiers, so that we don't need quotes


In my view, your example already proves the value.
The had to use `CallPattern`, `List`, `ExpressionCapture`, `type`, `Optional` for the same effect, without adding any actual value for readability (quite contrary, IMO).